src/fmeval/eval_algorithms/context_quality.py

```diff
@@ -0,0 +1,291 @@
+from fmeval.eval_algorithms.eval_algorithm import (
+    EvalAlgorithmInterface,
+    EvalAlgorithmConfig,
+)
+from fmeval.eval_algorithms.save_strategy import SaveStrategy
+from fmeval.eval_algorithms.util import get_dataset_configs, validate_dataset
+from fmeval.model_runners.bedrock_model_runner import BedrockModelRunner
+from fmeval.model_runners.model_runner import ModelRunner
+from fmeval.transforms.transform import Transform
+from fmeval.transforms.util import validate_call
+from fmeval.model_runners.composers.composers import PromptComposer
+from fmeval.transforms.transform_pipeline import TransformPipeline
+from fmeval.util import get_eval_results_path
+
+logger = logging.getLogger(__name__)
+
+CONTEXT_PRECISION_SCORE = "context_precision_score"
+
+DEFAULT_CONTEXT_PRECISION_PROMPT_TEMPLATE = (
+    "Given question, answer and context verify if the context was useful in "
+    "arriving at the given answer. Give verdict as 1 if useful and 0 if "
+    "not. question: $model_input, answer: $target_output, "
+    "context: $retrieved_context."
+    "The verdict should only contain an integer, either 1 or 0, do not give an explanation."
+)
+
+BINARY_SCORE_VALUES = ["0", "1"]
+
+# TODO: Pending judge model selection
+def get_default_judge_model() -> BedrockModelRunner:  # pragma: no cover
+    return BedrockModelRunner(
+        model_id="anthropic.claude-v2",
+        output="completion",
+        content_template='{"prompt": $prompt, "max_tokens_to_sample": 10000, "temperature": 0.1}',
+    )
+
+
+@dataclass(frozen=True)
+class ContextQualityConfig(EvalAlgorithmConfig):
+    """Configures the Context Quality evaluation algorithm.
+
+    :param target_output_delimiter: There can be multiple target outputs for a given input.
+        This delimiter is used to combine all retrieved contexts into a single string.
+    :param retrieved_context_delimiter: There can be multiple retrieved contexts for a given input.
+        This delimiter is used to combine all retrieved contexts into a single string.
+    """
+
+    target_output_delimiter: Optional[str] = "<OR>"
+    retrieved_context_delimiter: Optional[str] = "<AND>"
+
+
+class ContextQuality(EvalAlgorithmInterface):
+    eval_name = EvalAlgorithm.CONTEXT_QUALITY.value
+
+    def __init__(self, eval_algorithm_config: ContextQualityConfig = ContextQualityConfig()):
+        """ContextQuality initializer
+
+        :param eval_algorithm_config: Context Quality evaluation algorithm config.
+        """
+        super().__init__(eval_algorithm_config)
+        self.eval_algorithm_config = eval_algorithm_config
+
+    def evaluate(
+        self,
+        judge_model: Optional[ModelRunner] = None,
+        dataset_config: Optional[DataConfig] = None,
+        num_records: int = 100,
+        save: bool = False,
+        save_strategy: Optional[SaveStrategy] = None,
+        context_precision_prompt_template: Optional[str] = DEFAULT_CONTEXT_PRECISION_PROMPT_TEMPLATE,
```

Comment:
Where is this getting used?

Reply:
Good catch, yah I meant to move setting the default prompt template out of `ContextPrecisionScore` to `_build_pipeline`, will update.